### ISSUE TITLE
False positive for unused-import when only usage is inside of forward references

### ISSUE THREAD

**Issue by ringohoffman** [pylint-dev/pylint#9223]
### Bug description

```python
When the only usage of an import is in a [forward reference](https://peps.python.org/pep-0484/#forward-references), i.e. quoted type hints, pylint raises an `unused-import` warning.


from typing import Callable  # unused-import

from typing_extensions import TypeAlias

MyAlias: TypeAlias = "Callable[[int, int], int]"

def my_function(_: MyAlias) -> None:
    ...

my_function(1)
```

### Command used

```shell
pylint pylint_unused_import.py
```

### Pylint output

```python
pylint_unused_import.py:1:0: W0611: Unused Callable imported from typing (unused-import)
```

### Expected behavior

I expect for there to be no `unused-import` warning raised.

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.10.13
```
